ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -318,8 +335,8 @@ private static String idWatermark(CompactionInfo ci) {
   private boolean removeFiles(String location, ValidWriteIdList writeIdList, CompactionInfo ci)
       throws IOException, NoSuchObjectException, MetaException {
     Path locPath = new Path(location);
-    AcidDirectory dir = AcidUtils.getAcidState(locPath.getFileSystem(conf), locPath, conf, writeIdList, Ref.from(
-        false), false);
+    FileSystem fs = locPath.getFileSystem(conf);
```

Comment:
Independent change?

Reply:
no, created a separate cleanup method to handle `safe-delete` + extracted the common part